Scripts/BulkConversion/bulk_convert.ps1

```diff
@@ -0,0 +1,655 @@
+    Return $false
+}
+
+Function Test-RMConnection ($RemoteMachineName)
+{
+    ## Sends a network ping request to the Remote Machine
+    $PingResult = Test-Connection $RemoteMachineName -ErrorAction SilentlyContinue
+
+    ## Validates if a response of any kind has been returned.
+    If ($($PingResult.Count) -gt 0 )
+    {
+        ## If all Pings returned successfully, consider this a 100% good VM to work with.
+        If ($($PingResult.Count) -eq 4)
+        {
+            New-LogEntry -LogValue "Connection to $RemoteMachineName Successful." -Component "SharedScriptLib.ps1:Test-RMConnection"
+            Return $true
+        }
+
+        ## If some Pings were lost, still good, just potential network issue.
+        New-LogEntry -LogValue "Connection to $RemoteMachineName successful, Some packets were dropped." -Component "SharedScriptLib.ps1:Test-RMConnection" -Severity 2
+        Return $true
+    }
+
+    ## Returns false, no network response was available.
+    New-LogEntry -LogValue "Unable to Connect to $RemoteMachineName`r    - Ensure Firewall has been configured to allow remote connections and PING requests"  -Component "SharedScriptLib.ps1:Test-RMConnection" -Severity 3
+    Return $false
+}
+
+Function Compress-MSIXAppInstaller ($Path, $InstallerPath, $InstallerArgument)
+{
+    IF($Path[$Path.Length -1] -eq "\")
+        { $Path = $Path.Substring(0, $($Path.Length -1)) }
+
+    ## Identifies the original structure, creating a plan which can be used to restore after extraction
+    $Path               = $(Get-Item $Path).FullName
+    $ContainerPath      = $Path
+    $ExportPath         = "C:\Temp\Output"
+    $CMDScriptFilePath  = "$ContainerPath\MSIXCompressedExport.cmd"
+    $ScriptFilePath     = "$ContainerPath\MSIXCompressedExport.ps1"
+    $templateFilePath   = "$ContainerPath\MSIXCompressedExport.xml"
+    $EXEOutPath         = "$ContainerPath\MSIXCompressedExport.EXE"
+    $SEDOutPath         = "$ContainerPath\MSIXCompressedExport.SED"
+    $EXEFriendlyName    = "MSIXCompressedExport"
+    $EXECmdline         = $CMDScriptFilePath
+    $FileDetails        = @()
+    $XML                = ""
+    $SEDFiles           = ""
+    $SEDFolders         = ""
+    $FileIncrement      = 0
+    $DirectoryIncrement = 0
+
+    IF($(Get-Item $EXEOutPath -ErrorAction SilentlyContinue).Exists -eq $true)
+        { Remove-Item $EXEOutPath -Force }
+
+    New-LogEntry -LogValue "Multiple files required for app install compressing all content into a self-extracting exe" -Component "Compress-MSIXAppInstaller" -Severity 2 -WriteHost $VerboseLogging
+
+    ##############################  PS1  ##################################################################################
+    ## Creates the PowerShell script which will export the contents to proper path and trigger application installation. ##
+    $($ScriptContent = @'
+$XMLData = [xml](Get-Content -Path ".\MSIXCompressedExport.xml")
```

Comment:
Nit: indentation

Reply:
Can't be indented. Limitation of the value type.

Reply:
This can't be indented. Limitation of the value type.